code/modules/mob/living/carbon/slime/slime.dm

```diff
@@ -894,6 +894,62 @@ mob/living/carbon/slime/var/temperature_resistance = T0C+75
 	icon_state = "bottle14"
 	w_class = W_CLASS_TINY
 
+/obj/item/weapon/slimechange
+	name = "slime dye"
+	desc = "A bottle of dye that can be used on a slime to change it's color."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "bottle14"
+	w_class = W_CLASS_TINY
+
+/obj/item/weapon/slimechange/attack(var/mob/living/M, mob/user as mob)
+	if(istype(M, /mob/living/carbon/slime))
+		var/mob/living/carbon/slime/OS = M
+		if(OS.stat)
+			to_chat(user, "<span class='warning'>You're unable to use the [src] on the dead slime!</span>")
+			return..()
+		var/new_color = input(user, "Choose a new color for the dye:", "Color Select") as anything in list("adamantine", "black", "blue", "bluespace", "cerulean", "darkpurple", "darkblue", "darkpurple", "gold", "grey", "lightpink", "metal", "oil", "orange", "pink", "purple", "pyrite", "red", "sepia", "silver", "yellow")
+		if(new_color)
```

Comment:
Not sure this takes into account if the slime moves away

Reply:
While I can understand it being a possible concern of sanity it's a non-issue.

Reply:
Yes it is an issue.

Reply:
None of the other slime potions check to see if the slime is still next to them at the end of the potion use like the pet one. I assume this is because the slimes are mobs that will frequestly walk away from you while you're selecting/working in the input.

Reply:
This one asks for an input, which can be held off for a while.

You can slap the slime with the potion while they're alive, have somebody else kill them, then 'revive' the slime by pressing submit.

Reply:
All right point taken on that issue I will add some sanity there